From aceb2c1cc6a7bbdbb36ba1b074978adf6c49a04f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Konrad=20Zapa=C5=82owicz?=
 <konrad.zapalowicz@chainguard.dev>
Date: Fri, 5 Sep 2025 11:31:46 +0200
Subject: [PATCH 4/4] refactor(entitlement): always pick default client
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The entitlements client is determined dynamically, based on the state of the
environment variables, and can be either tied to Control Plane or AWS
Marketplace.

The chainguard images are not served through the AWS Marketplace nevertheless
if a depl;oyment would have the MARKETPLACE_TYPE env var set to the appropriate
value the AWS entitlement client would be picked.

We do not want that and in order to mitigate this commit is making sure that
the entitlement client is always the default one regardless of the state of the
marketplace env variable.

Note that ignoring the marketplace env variable will allow for seamless
deploymenmts where the flux-operator has already been deployed through the AWS
Marketplace.

Signed-off-by: Konrad Zapałowicz <konrad.zapalowicz@chainguard.dev>
---
 internal/entitlement/client.go | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/internal/entitlement/client.go b/internal/entitlement/client.go
index ebb91af..fa8fdd6 100644
--- a/internal/entitlement/client.go
+++ b/internal/entitlement/client.go
@@ -6,8 +6,6 @@ package entitlement
 import (
 	"context"
 	"fmt"
-	"os"
-	"strings"
 )
 
 const (
@@ -42,14 +40,9 @@ type Client interface {
 	GetVendor() string
 }
 
-// NewClient returns a new entitlement client based on the
-// marketplace type environment variable.
+// NewClient returns a DefaultClient with DefaultVendor no matter what.
 func NewClient() (Client, error) {
 	vendor := DefaultVendor
-	marketplace, found := os.LookupEnv(MarketplaceTypeEnvKey)
-	if found && marketplace != "" && marketplace != DefaultVendor {
-		vendor = fmt.Sprintf("%s-%s", DefaultVendor, strings.ToLower(marketplace))
-	}
 
 	switch vendor {
 	case DefaultVendor:
-- 
2.50.1

